src/py42/services/_connection.py

```diff
@@ -191,16 +191,25 @@ def request(
                 proxies=proxies or settings.proxies,
             )
 
-            if not stream and response is not None:
-                # setting this manually speeds up read times
-                response.encoding = u"utf-8"
-
-            if response is not None and 200 <= response.status_code <= 399:
-                return Py42Response(response)
-
-            if response is not None and response.status_code == 401:
-                if isinstance(self._auth, C42RenewableAuth):
-                    self._auth.clear_credentials()
+            if response is not None:
+                if not stream:
+                    # setting this manually speeds up read times
+                    response.encoding = u"utf-8"
+
+                debug.logger.error(
+                    u"Response: status={}, data={}".format(
+                        response.status_code, response.text
```

Comment:
If the response is a stream, accessing response.text here blocks to consume the stream, defeating the purpose of streaming.

Reply:
Refactored, logs response when response is not a stream! Do we want to log stream response too?

Reply:
Yeah we should log the response line whether it's a stream or not, but if it is a stream, instead of accessing `response.text` and consuming the stream, we should swap it out with `<streamed>` or something to indicate that it's a streamed response that may not be consumed yet.

Reply:
added.